services/watchcat/watchcat_init.c

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ * Copyrigth (c) 2019 by Moritz Wenk <max-1973@gmx.de>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License (either version 2 or
+ * version 3) as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ *
+ * For more information on the GPL, please go to:
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include <avr/pgmspace.h>
+#include <string.h>
+
+#include "config.h"
+#include "core/debug.h"
+#include "core/portio/portio.h"
+#include "protocols/uip/uip.h"
+#include "watchcat.h"
+#include "watchcat_init.h"
+
+extern struct VirtualPin vpin[IO_PORTS];
```

Comment:
Is it a good idea to extract the initialization to a separate file and open the visibility of the variables? What is the added value?